File: 2442-number-of-arithmetic-triplets/2442-number-of-arithmetic-triplets.cpp
class Solution {
public:
    int arithmeticTriplets(vector<int>& arr, int diff) {
        int n = arr.size();
        unordered_map<int,int> mp;

        for(int i=0;i<n;i++){
            mp[arr[i]]++;
        }
        int cnt = 0;
        for(int j = 1;j<n;j++){
            if(mp.find(arr[j] - diff) != mp.end() && mp.find(arr[j] + diff) != mp.end()){
                cnt++;
            }

        }
        return cnt;
    }
};

File: 2442-number-of-arithmetic-triplets/2442-number-of-arithmetic-triplets.py
class Solution(object):
    def arithmeticTriplets(self, nums, diff):
       
        s = set(nums)   # automatically handles uniqueness
        cnt = 0
        for i in range(1, len(nums) - 1):
            if (nums[i] - diff) in s and (nums[i] + diff) in s:
                cnt += 1
        return cnt
